### ISSUE TITLE
touchscreen click events are not detected properly (on Linux)

### ISSUE THREAD

**Issue by girtsf** [emilk/egui#4895]
**Describe the bug**

With any widget, such as a `Button` or  `Label`, `.clicked()` events are not recognized when the screen is tapped once. Instead, a single tap triggers `long_touched()` after a delay. Tapping twice in a quick succession generates a `clicked()` followed by a `long_touched()`.

**To Reproduce**

`Cargo.toml`:

```
[package]
name = "repro"
version = "0.1.0"
edition = "2021"

[dependencies]
egui = { git = "https://github.com/emilk/egui", branch = "master" }
eframe = { git = "https://github.com/emilk/egui", branch = "master" }
```

`src/main.rs`:

```
#[derive(Default)]
struct App {
    log: String,
}

impl eframe::App for App {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::CentralPanel::default().show(ctx, |ui| {
            let button = ui.button("click me");
            if button.clicked() {
                self.log.push_str("clicked()\n");
            }
            if button.long_touched() {
                self.log.push_str("long_touched()\n");
            }
            ui.label(&self.log);
        });
    }
}

fn main() {
    eframe::run_native(
        "test",
        eframe::NativeOptions::default(),
        Box::new(|_cc| Ok(Box::new(App::default()))),
    )
    .unwrap();
}
```

1. `cargo run` the above code
2. tap on "click me" button

**Expected behavior**
`clicked()` gets shown on the screen.

**Screenshots**
![Screenshot from 2024-07-31 14-48-42](https://github.com/user-attachments/assets/c9fe5eb0-071d-4063-9287-fffe384b3098)

This can also be reproduced by `egui_demo_app`, built from latest `master` (d856f7b3a58fc65a81ff9657f5c6e1ab462b7bb6), by opening "Input Test" and tapping on "Sense::click" label:

![Screenshot from 2024-07-31 14-53-00](https://github.com/user-attachments/assets/b0723de5-e6b1-438e-b85d-f673480d5595)

**Desktop:**
 - OS: Ubuntu 24.04
 - Device:  Lenovo Yoga 7 2-in-1 14IML9
 - Wayland

(I tested on Windows, on a slightly different touchscreen-equipped Lenovo laptop, and it works correctly there.)

**Additional context**

This might be related to the observations found in #4668 ? 

I bisected the behavior using `egui_demo_app`: last working commit is 8503a851138c36d76289202d603c23229558b890, and it breaks at 8e5959d55d2f5799a365dd2474b8868b7d49ad14.

**Comment by Zoxc** [comment]
I see similar behavior on Android with 0.28.1.

**Comment by gfolkmanis-nhgh** [comment]
I did some further digging and found that we were not getting `WindowEvent::Touch` events from `winit` with `TouchPhase::Ended` when finger was lifted from the screen. Instead, the event would get sent together with the next `TouchPhase::Started`. 

I confirmed the behavior happens with `winit` example code (no egui involved) on v0.30.2. Tested later versions and found that the issue was fixed in `winit` v0.30.4.

For funsies, I kept digging into `winit` changes, and it's likely that bumping to a newer `smithay-client-toolkit` fixed it (which brought in https://github.com/Smithay/client-toolkit/commit/8708dd5d8b8972bbd8cfe0ffa7ae649d42115333). I didn't retest to confirm the exact root cause, but that's likely it.

So once `egui` switches to `winit` `v0.30.4` or later, this issue should resolve itself. Looks like it is currently pinned to `v0.30.2` as per https://github.com/emilk/egui/pull/4849. Looks like the fix for that might be ready to go as of today, and https://github.com/emilk/egui/pull/4939 might bring it in!
